frontend/Auth/FinishResetPassword.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
+import { Button, ButtonKind } from "../Styleguide/Button";
+import { FormField } from "../Styleguide/FormField";
+import { FormFieldLabel } from "../Styleguide/FormFieldLabel";
+import { Input } from "../Styleguide/Input";
+import { RouterProps } from "react-router";
+import { useMutation, useQuery } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function FinishResetPassword(props: RouterProps) {
+  const [finishResetPasswordFormData, setFinishResetPasswordFormData] =
+    useState<FinishResetPasswordFormData>({
+      password: "",
+    });
+
+  const paramString = props.history.location.search;
+  let searchParams = new URLSearchParams(paramString);
+  const token = searchParams.get("jwt") || "MISSING TOKEN";
+
+  const queryFunctionhelper = (queryKey) => {
+    const token = queryKey["queryKey"][0];
+    return flaxFetch<object>(
+      `/validate-token/${token}?tokenType=passwordReset`,
+      {
+        method: "GET",
+      }
+    );
+  };
+
+  const info = useQuery(token, queryFunctionhelper);
+
+  const submitMutation = useMutation<
+    void,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >((evt) => {
+    evt.preventDefault();
+    let requestPromise = flaxFetch<void>(`/send-reset-password-email`, {
+      method: "POST",
+      body: {
+        password: finishResetPasswordFormData.password,
+        token: token,
+      },
+    });
+    return requestPromise;
+  });
+
+  return (
+    <div className="flex justify-center h-screen p-24 sm:pt-80">
+      <form
+        onSubmit={unary(submitMutation.mutate)}
+        className="relative lg:max-w-sm"
+      >
+        <h1 className="text-gray-500 place-self-start mb-2 text-5xl lg:text-xl">
+          Set New Password
+        </h1>
+        <p className="text-left text-gray-600 py-8 text-4xl lg:text-sm">
+          Enter your new password below.
```

Comment:
I wonder if it would be helpful to have an input for the email, but then have it filled in and disabled. That way they'd know which account they're setting a password for. To do that, we'd need to add the `userEmail` to the backend api's response, which would involve getting the user id from the JWT model and then getting the user object for that user id.

I think showing this would be a nice touch, let's add it.

Reply:
This is a great idea! I will add it.